src/main/webapp/student/enrollment/schoolClass/schoolClassesSelection.jsp

```diff
@@ -0,0 +1,116 @@
+					
+						<bean:define id="link">/schoolClassStudentEnrollment.do?method=viewSchoolClass&schoolClassID=<c:out value="${schoolClass.externalId}" />&registrationID=<c:out value="${enrollmentBean.registration.externalId}" />&enrolmentPeriodID=<c:out value="${enrollmentBean.enrolmentPeriod.externalId}" /></bean:define>
+						<html:link page="<%= link %>">
+							<c:out value="${schoolClass.editablePartOfName}" />
+							<c:if test="${(not empty currentSchoolClass) and (schoolClass eq currentSchoolClass)}">
+								<span class="badge"><bean:message bundle="STUDENT_RESOURCES" key="label.schoolClassStudentEnrollment.selected" /></span>
+							</c:if>
+						</html:link>
+						
+					</li>
+				</c:forEach>
+			</ul>
+			
+			<p></p>
+			
+			<c:if test="${not empty schoolClassToDisplay}">
+				<c:choose>
+				    <c:when test="${(not empty currentSchoolClass) and (schoolClassToDisplay eq currentSchoolClass)}">
+						<bean:define id="removeSchoolClassLink">/schoolClassStudentEnrollment.do?method=enrollInSchoolClass&registrationID=<c:out value="${enrollmentBean.registration.externalId}" />&enrolmentPeriodID=<c:out value="${enrollmentBean.enrolmentPeriod.externalId}" /></bean:define>
+						<html:link page="<%= removeSchoolClassLink %>" styleClass="btn btn-warning btn-xs mtop15">
+							<span class="glyphicon glyphicon-remove" aria-hidden="true"></span> <bean:message bundle="STUDENT_RESOURCES" key="button.schoolClassStudentEnrollment.unselectSchoolClass" />
+						</html:link>			
+				    </c:when>
+				    <c:otherwise>
+						<bean:define id="selectSchoolClassLink">/schoolClassStudentEnrollment.do?method=enrollInSchoolClass&schoolClassID=<c:out value="${schoolClassToDisplay.externalId}" />&registrationID=<c:out value="${enrollmentBean.registration.externalId}" />&enrolmentPeriodID=<c:out value="${enrollmentBean.enrolmentPeriod.externalId}" /></bean:define>
+						<bean:define id="selectSchoolClassLinkCssClass">btn btn-primary btn-xs mtop15 <c:if test="${not enrollmentBean.schoolClassToDisplayFree}">disabled</c:if></bean:define>
+						<html:link page="<%= selectSchoolClassLink %>" styleClass="<%= selectSchoolClassLinkCssClass %>">
+							<span class="glyphicon glyphicon-ok" aria-hidden="true"></span> <bean:message bundle="STUDENT_RESOURCES" key="button.schoolClassStudentEnrollment.selectSchoolClass" />
+						</html:link>		    
+				    	<c:if test="${not enrollmentBean.schoolClassToDisplayFree}"><span class="text-warning"><bean:message bundle="STUDENT_RESOURCES" key="label.schoolClassStudentEnrollment.fullSchoolClass" /></span></c:if>
+				    </c:otherwise>
+				</c:choose>
+		
+				<c:set value="${enrollmentBean.schoolClassToDisplayLessons}" var="schoolClassToDisplayLessons"/>
+				<c:if test="${not empty schoolClassToDisplayLessons}">
+					<div class="mtop15">
+						<app:gerarHorario name="schoolClassToDisplayLessons" type="<%= TimeTableType.CLASS_TIMETABLE %>" application="<%= request.getContextPath() %>"/>
```

Comment:
Have you considered using fullcalendar as we have been doing for new features? This old timetable component is static, badly styled, full of legacy dependencies like infoObjects, and is intended for deprecation.

You have usage examples here: https://github.com/FenixEdu/fenixedu-learning/blob/master/src/main/webapp/themes/fenixedu-learning-theme/calendarEvents.html